torchrl/envs/transforms/transforms.py

```diff
@@ -4109,74 +4109,158 @@ class RewardSum(Transform):
+        # elif isinstance(done_keys, list) and len(done_keys) == 1:
+        #     # If there is one done key than it is used for all rewards
+        #     self._done_keys = done_keys * len(in_keys)
+        # elif isinstance(done_keys, list) and len(done_keys) == len(in_keys):
+        #     # The full list of done_keys has been provided
+        #     self._done_keys = done_keys
+        # else:
+        #     raise ValueError(
+        #         f"Provided done_keys should be a list of length 1 or length equals to the number of in_keys, got {done_keys}"
+        #     )
+        #
+        super().__init__(in_keys=in_keys, out_keys=out_keys)
+        self._reset_keys = reset_keys
+
+    @property
+    def in_keys(self):
+        in_keys = self.__dict__.get("_in_keys", None)
+        if in_keys in (None, []):
+            # retrieve rewards from parent env
+            parent = self.parent
+            if parent is None:
+                # fall back on in_keys = ["reward"]
+                in_keys = ["reward"]
+                # or this?
+                # raise TypeError("in_keys not provided but parent env not found. "
+                #                 "Make sure that the in_keys (reward) are provided during "
+                #                 "construction if the transform does not have a container env.")
+            else:
+                in_keys = copy(parent.reward_keys)
+            self._in_keys = in_keys
+        return in_keys
+
+    @in_keys.setter
+    def in_keys(self, value):
+        self._in_keys = value
+
+    @property
+    def out_keys(self):
+        out_keys = self.__dict__.get("_out_keys", None)
+        if out_keys in (None, []):
+            out_keys = [
+                _replace_last(in_key, f"episode_{_unravel_key_to_tuple(in_key)[-1]}")
+                for in_key in self.in_keys
+            ]
+            self._out_keys = out_keys
+        return out_keys
+
+    @out_keys.setter
+    def out_keys(self, out_keys):
+        # we must access the private attribute because this check occurs before
+        # the parent env is defined
+        if len(self._in_keys) != len(out_keys):
+            raise ValueError(
+                "RewardSum expects the same number of input and output keys"
             )
+        self._out_keys = out_keys
 
-        super().__init__(in_keys=in_keys, out_keys=out_keys)
+    @property
+    def reset_keys(self):
+        reset_keys = self.__dict__.get("_reset_keys", None)
+        if reset_keys is None:
+            parent = self.parent
+            if parent is None:
+                raise TypeError(
+                    "reset_keys not provided but parent env not found. "
+                    "Make sure that the reset_keys are provided during "
+                    "construction if the transform does not have a container env."
+                )
+            reset_keys = copy(parent.reset_keys)
+            self._reset_keys = reset_keys
+        return reset_keys
```

Comment:
when we are using the default env keys, are we sure that `env.reward_keys` and `env.reset_keys` are in the right order? (aka if i zip the 2 lists i will get the right corresponding keys)

Reply:
Yes they are! But it's a good point